dGame/dBehaviors/BehaviorContext.cpp

```diff
@@ -289,46 +290,74 @@ void BehaviorContext::Reset() {
 	this->scheduledUpdates.clear();
 }
 
-std::vector<LWOOBJID> BehaviorContext::GetValidTargets(int32_t ignoreFaction, int32_t includeFaction, bool targetSelf, bool targetEnemy, bool targetFriend) const {
-	auto* entity = EntityManager::Instance()->GetEntity(this->caster);
+std::vector<Entity*> BehaviorContext::FilterTargets(std::forward_list<int32_t> ignoreFactionList, std::forward_list<int32_t> includeFactionList, bool targetSelf, bool targetEnemy, bool targetFriend, bool targetTeam) const {
+	std::vector<Entity*> targets = {};
 
-	std::vector<LWOOBJID> targets;
+	// if we aren't targeting anything, then return empty targets list
+	if (!targetSelf && !targetEnemy && !targetFriend && !targetTeam && ignoreFactionList.empty() && includeFactionList.empty()) return targets;
 
-	if (entity == nullptr) {
-		Game::logger->Log("BehaviorContext", "Invalid entity for (%llu)!", this->originator);
+	auto* caster = EntityManager::Instance()->GetEntity(this->caster);
 
+	// if the caster is not there, return empty targets list
+	if (!caster) {
+		Game::logger->LogDebug("BehaviorContext", "Invalid caster for (%llu)!", this->originator);
 		return targets;
 	}
 
-	if (!ignoreFaction && !includeFaction) {
-		for (auto entry : entity->GetTargetsInPhantom()) {
-			auto* instance = EntityManager::Instance()->GetEntity(entry);
-
-			if (instance == nullptr) {
-				continue;
+	auto* casterDestroyableComponent = caster->GetComponent<DestroyableComponent>();
+	if (!casterDestroyableComponent) return targets;
+
+	auto candidates = EntityManager::Instance()->GetEntitiesByComponent(eReplicaComponentType::DESTROYABLE);
```

Comment:
This should actually be done by getting all entities in the radius in the AOE component (capping out at a radius of 1000.0f) and then filter on those targets, but this is how DLU was doing it before and it wasn't visible affecting performance.
Will probably fix this later, but it's fine as it for now